source/slang/slang-compiler.h

```diff
@@ -1564,8 +1568,11 @@ namespace Slang
     class FrontEndCompileRequest : public CompileRequestBase
     {
     public:
+            /// Note that writers can be parsed as nullptr to disable output,
+            /// and individual channels set to null to disable them
         FrontEndCompileRequest(
             Linkage*        linkage,
+            StdWriters*     writers, 
```

Comment:
I'm not a fan of having `StdWriters` reach its way deeper into the Slang library, because it creates a potential for confusion about "where should I send this output?".

Ideally we should only have two cases of output:

1. Diagnostic output. This is always written to a diagnostic sink, which emits to `stderr` for `slangc` and can be captured into buffers for API-based use.
2. File output ("compile results"). These are always captures into buffers for API-based used, and for `slangc` are always either written to filesystem paths or `stdout`.

The redirection feature really only exists to support the test-runner using in-process execution while still running Slang in "`slangc` mode." I feel like we need to push back and start solving more of this at the test-runner level when we next get time to work on the test system.

Reply:
I don't think it is worth trying to change anything for this PR, since none of this is really related to the `__LINE__`/`__FILE__` feature.